AudioManagerUtilities/src/CAmCommonAPIWrapper.cpp

```diff
@@ -243,10 +260,12 @@ void CAmCommonAPIWrapper::registerWatch(CommonAPI::Watch* watch, const CommonAPI
     am_Error_e error = mpSocketHandler->addFDPoll(pollfd_.fd, pollfd_.events, &pCommonPrepareCallback, &pCommonFireCallback, &pCommonCheckCallback, &pCommonDispatchCallback, watch, handle);
 
     //if everything is alright, add the watch and the handle to our map so we know this relationship
-    if (error == !am_Error_e::E_OK || handle == 0)
+    if (error != am_Error_e::E_OK || handle == 0)
```

Comment:
Even without knowing the code very well this new code definitely looks more correct...